### PR TITLE
NAS-116040 / 22.12 / Nas 116040 Clean up SysInfoComponent

### PR DESCRIPTION
**Pull request by Raz-Dva**
Check SysInfoComponent.
System Settings -> General ( Support widget )

### CODE REVIEW

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.ts [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
-
-  // Additional fields for licensed customers
-  @Input() customer_name: string;
-  @Input() features: string[];
-  @Input() contract_type: string;
-  @Input() expiration_date: string;
-  @Input() add_hardware: string;
-  @Input() daysLeftinContract: number;
+  @Input() has_license: boolean;
+  @Input() license_info: LicenseInfoInSupport;
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.ts:11]: need to be camelCase.

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.html [hunk 2]
```diff
@@ -1,47 +1,65 @@
...
+    <span class="label">{{'Contract Type' | translate}}:</span>
+    <span class="value">{{ license_info.contract_type}}</span>
   </div>
-  <div class="sys-info-line"><span class="label"> {{"Memory:" | translate}}</span> 
-    <span class="value">{{memory}}</span>
+  <div class="sys-info-row">
+    <span class="label">{{'Expiration Date' | translate}}:</span>
+    <span class="value">{{ license_info.expiration_date}}</span>
+    <span *ngIf="license_info.daysLeftinContract && license_info.daysLeftinContract > 0" class="value">
+      ( {{'expires in' | translate}} {{ '{n, plural, one {# day} other {# days} }' | translate: {n: license_info.daysLeftinContract} }}
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.html:18]: Needs to be one string: 'expires in {n...}'

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.html [hunk 3]
```diff
@@ -1,47 +1,65 @@
...
+    </span>
+    <span *ngIf="license_info.daysLeftinContract < 0 " class="value">
+      ( {{'EXPIRED' | translate}} )
+    </span>
   </div>
-  <div *ngIf="serial" class="fn-sys-info-line"><span class="label"> {{"Serial Number:" | translate}}</span> 
-    <span class="value">{{serial}}</span>
+  <div class="sys-info-row">
+    <span class="label">{{'Model' | translate}}:</span>
+    <span class="value">{{ system_info.model}}</span>
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.html:30]: Inconsistent spacing around {{ }}. Should be with spaces.

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.html [hunk 4]
```diff
@@ -1,47 +1,65 @@
...
+  <div class="sys-info-row">
+    <span class="label">{{'Contract Type' | translate}}:</span>
+    <span class="value">{{ license_info.contract_type}}</span>
   </div>
-  <div class="sys-info-line"><span class="label"> {{"Memory:" | translate}}</span> 
-    <span class="value">{{memory}}</span>
+  <div class="sys-info-row">
+    <span class="label">{{'Expiration Date' | translate}}:</span>
+    <span class="value">{{ license_info.expiration_date}}</span>
+    <span *ngIf="license_info.daysLeftinContract && license_info.daysLeftinContract > 0" class="value">
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.html:17]: First part of if statement seems unnecessary.

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.ts [hunk 5]
```diff
@@ -1,23 +1,13 @@
 import { Component, Input } from '@angular/core';
+import { LicenseInfoInSupport } from 'app/pages/system/general-settings/support/license-info-in-support.interface';
+import { SystemInfoInSupport } from 'app/pages/system/general-settings/support/system-info-in-support.interface';
 
 @Component({
   selector: 'ix-sys-info',
   templateUrl: './sys-info.component.html',
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.ts:7]: Please add changeDetection.onPush.

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    add_hardware: 'NONE',
+  };
+
+  let spectator: SpectatorHost<SysInfoComponent>;
+  const createHost = createHostFactory({
+    component: SysInfoComponent,
+  });
+
+  beforeEach(() => {
+    spectator = createHost('<ix-sys-info [system_info]="systemInfo"></ix-sys-info>', {
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.spec.ts:28]: You don't need to use createHost here. You can use normal createComponent, there is a second argument in it that allows to pass in Inputs.

File: src/app/pages/system/general-settings/support/sys-info/sys-info.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      hostProps: {
+        systemInfo,
+        has_license: false,
+      },
+    });
+  });
+
+  it('it shows a block with system info', () => {
+    const sysInfoItems = spectator.queryAll('.sys-info-wrapper .value');
```
Comment (undsoft) [inline src/app/pages/system/general-settings/support/sys-info/sys-info.component.spec.ts:37]: would be good to also test labels. You could use reduce to form a dictionary: { Label1: Value, Label2: Value }, etc.

**Review by undsoft — CHANGES_REQUESTED** [review]
Looks good overall.

**Comment by undsoft** [comment]
Some machines may have two serial numbers. There is a bug that doesn't show them properly.
It was fixed here: https://github.com/truenas/webui/pull/6796 
Can you also incorporate this fix in your code?

**Comment by Raz-Dva** [comment]
Yea. I added changes from https://github.com/truenas/webui/pull/6796

**Review by undsoft — APPROVED** [review]
Looks good.

**Comment by Raz-Dva** [comment]
Thanks.

**Pull request merged**